### ISSUE TITLE
Flutter web Stuck at Hot restart

### ISSUE THREAD

**Issue by irajeshh** [flutter/flutter#98393]
**Tried on chrome, Brave, Edge. Also tried on old & fresh projects.**
Also I turned of auto hotreload, hotrestart on save. Then Pressed the hotreload button manually. Nothing worked!

## Steps to Reproduce

1. Create a fresh flutter project on windows.
2. Add web support.

**Expected results:** Updating the implemented changes

**Actual results:** Stuck on what it was before
![image](https://user-images.githubusercontent.com/85046336/153839762-40c66ebf-fe1c-4e84-a4ed-7e436247f709.png)

<details>
<summary>Code sample</summary>

```
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData.dark(),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  // This widget is the home page of your application. It is stateful, meaning
  // that it has a State object (defined below) that contains fields that affect
  // how it looks.

  // This class is the configuration for the state. It holds the values (in this
  // case the title) provided by the parent (in this case the App widget) and
  // used by the build method of the State. Fields in a Widget subclass are
  // always marked "final".

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      // This call to setState tells the Flutter framework that something has
      // changed in this State, which causes it to rerun the build method below
      // so that the display can reflect the updated values. If we changed
      // _counter without calling setState(), then the build method would not be
      // called again, and so nothing would appear to happen.
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    // This method is rerun every time setState is called, for instance as done
    // by the _incrementCounter method above.
    //
    // The Flutter framework has been optimized to make rerunning build methods
    // fast, so that you can just rebuild anything that needs updating rather
    // than having to individually change instances of widgets.
    return Scaffold(
      appBar: AppBar(
        // Here we take the value from the MyHomePage object that was created by
        // the App.build method, and use it to set our appbar title.
        title: const Text("Welcome3"),
      ),
      body: Center(
        // Center is a layout widget. It takes a single child and positions it
        // in the middle of the parent.
        child: Column(
          // Column is also a layout widget. It takes a list of children and
          // arranges them vertically. By default, it sizes itself to fit its
          // children horizontally, and tries to be as tall as its parent.
          //
          // Invoke "debug painting" (press "p" in the console, choose the
          // "Toggle Debug Paint" action from the Flutter Inspector in Android
          // Studio, or the "Toggle Debug Paint" command in Visual Studio Code)
          // to see the wireframe for each widget.
          //
          // Column has various properties to control how it sizes itself and
          // how it positions its children. Here we use mainAxisAlignment to
          // center the children vertically; the main axis here is the vertical
          // axis because Columns are vertical (the cross axis would be
          // horizontal).
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headline4,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```

```dart
```

</details>

<details>
  <summary>Logs</summary>

```
```

```
Analyzing webtest...                                                    
No issues found! (ran in 10.1s)
```

```
[√] Flutter (Channel stable, 2.10.1, on Microsoft Windows [Version 10.0.19043.1415], locale en-IN)
[√] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.0.6)   
[√] Android Studio (version 2021.1)
[√] VS Code, 64-bit edition (version 1.64.2)
[√] Connected device (3 available)
[√] HTTP Host Availability
```

</details>

**Comment by maheshmnj** [comment]
Hi @irajeshh, Thanks for filing the issue. I am unable to reproduce the issue does this happen frequently or is it a rare case? Also, do you see this happening only in vscode or when running from other IDE or terminal too?

**Comment by irajeshh** [comment]
@maheshmnj Today I tried the same project on Zorin Os.
1) **In Vscode** is not launching, it stucks at `Waiting for connection from debug service on Chrome`
2) In Android Studio It works as expected.

**Here is the flutter doctor logs**
```
[✓] Flutter (Channel stable, 2.8.1, on Zorin OS 16 5.13.0-27-generic, locale
    en_IN)
[!] Android toolchain - develop for Android devices (Android SDK version
    32.1.0-rc1)
    ✗ cmdline-tools component is missing
      Run `path/to/sdkmanager --install "cmdline-tools;latest"`
      See https://developer.android.com/studio/command-line for more details.
    ✗ Android license status unknown.
      Run `flutter doctor --android-licenses` to accept the SDK licenses.
      See https://flutter.dev/docs/get-started/install/linux#android-setup for
      more details.
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.63.2)
[✓] Connected device (2 available)

! Doctor found issues in 1 category.
```
I'll update the flutter sdk version to `2.10.1` like in windows, then I'll update the issue.

**Comment by irajeshh** [comment]
Iv'e updated the flutter version to 2.10.1, but the same `Waiting for connection from debug service on Chrome` stucking happened on vscode. But Android studio worked as before.
The issue is on Windows platform only.

**Comment by irajeshh** [comment]
On `windows` platform, both `Vscode` & `Android Studio` is making the same `Performing hot restart...` issue. But Find out a step that shows the code change on Broswer:
**Step1:** I have to make some change in the code
**Step2:** Press HotReload button on vscode.
**Step3:** **Immediately** refresh chrome by using `**Ctrl+R**` (on browser window)

**Comment by irajeshh** [comment]
On Linux plaform, I'v fixed the `Waiting for connection from debug service on Chrome` bug by Uninstalling brave browser [I installed it from snap store on zorin os]. Then re-installed it again with `apt`. and set the chrome executable value in ~bashrc file.
But still the problem exists on `windows` platform.

**Comment by maheshmnj** [comment]
@irajeshh, I am unable to reproduce the issue, Hot Restart completes instantly with your [reproducible steps
](https://github.com/flutter/flutter/issues/98393#issuecomment-1039997536)

```
Restarted application in 160ms.
Restarted application in 437ms.
Restarted application in 273ms.
Reload already in progress, ignoring request
Restarted application in 5,556ms.
Restarted application in 315ms.
Reload already in progress, ignoring request
Reload already in progress, ignoring request
Restarted application in 7,755ms.
ChromeProxyService: Failed to evaluate expression 'entrypoint.main': InternalError: No frame with index 150.
Error
Error
Restarted application in 2,871ms.
Restarted application in 282ms.
Restarted application in 4,408ms.
Restarted application in 301ms.

``` 

<details>
<summary>flutter doctor -v</summary>

```
[√] Flutter (Channel stable, 2.10.1, on Microsoft Windows [Version 10.0.19044.1526], locale en-US)
    • Flutter version 2.10.1 at C:\flutter_sdk\stable
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision db747aa133 (6 days ago), 2022-02-09 13:57:35 -0600
    • Engine revision ab46186b24
    • Dart version 2.16.1
    • DevTools version 2.9.2

[√] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at C:\Users\mahesh\AppData\Local\Android\sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java      
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)    
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.11.4)      
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.11.31727.386
    • Windows 10 SDK version 10.0.19041.0

[√] Android Studio (version 2020.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[√] VS Code (version 1.64.2)
    • VS Code at C:\Users\mahesh\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.34.0

[√] Connected device (3 available)
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.19044.1526]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 98.0.4758.102
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 97.0.1072.76

[√] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```
</details>

Are there any logs /errors in the console when the hot restart is stuck?

**Comment by irajeshh** [comment]
I think the issue is already [here](https://github.com/flutter/flutter/issues/91407). And I just stopped using windows, so closing this issue here.

**Comment by maheshmnj** [comment]
The issue you linked was reproducible in 2.5 and was fixed in the consecutive stable release (2.8), In case you have any further info or logs on the reproduction feel free to write back on this, issue or file a new issue.
